server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -454,7 +459,8 @@ public ClusterState execute(ClusterState currentState) {
                             if (metadata.customs() != null) {
                                 for (ObjectObjectCursor<String, Metadata.Custom> cursor : metadata.customs()) {
                                     if (RepositoriesMetadata.TYPE.equals(cursor.key) == false
-                                            && DataStreamMetadata.TYPE.equals(cursor.key) == false) {
+                                            && DataStreamMetadata.TYPE.equals(cursor.key) == false
+                                            && (operatorPrivilegesEnabled == false || "autoscaling".equals(cursor.key) == false)) {
```

Comment:
Rather than switch on the "autoscaling" key I would prefer to add a marker interface "NotRestorableCustomMetadata", implemented by `AutoscalingMetadata` and use instanceof to omit it instead.

Reply:
Updated. Thanks!